pkg/types/helper.go

```diff
@@ -55,7 +55,7 @@ func Round(f float64, dec int) float64 {
 func Truncate(f float64, dec int) float64 {
 	shift := math.Pow10(dec)
 	tmp := f * shift
-	if math.IsInf(tmp, 0) {
+	if math.IsInf(tmp, 0) || math.IsNaN(tmp) {
```

Comment:
Better add comments here to explain how tmp be NAN while shift is not.

Reply:
> Better add comments here to explain how tmp be NAN while shift is not.

added